hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/common/table/read/TestDeleteRecordLogic.scala

```diff
@@ -199,23 +200,27 @@ class TestDeleteRecordLogic extends SparkClientFunctionalTestHarness{
 object TestDeleteRecordLogic {
   def provideParams(): java.util.List[Arguments] = {
     java.util.Arrays.asList(
-      Arguments.of("false", "COPY_ON_WRITE", "AVRO"),
-      Arguments.of("false", "COPY_ON_WRITE", "SPARK"),
-      Arguments.of("false", "MERGE_ON_READ", "AVRO"),
-      Arguments.of("false", "MERGE_ON_READ", "SPARK"),
-      Arguments.of("true", "COPY_ON_WRITE", "AVRO"),
-      Arguments.of("true", "COPY_ON_WRITE", "SPARK"),
-      Arguments.of("true", "MERGE_ON_READ", "AVRO"),
-      Arguments.of("true", "MERGE_ON_READ", "SPARK"))
+      Arguments.of("false", "COPY_ON_WRITE", "AVRO", "false"),
+      Arguments.of("false", "COPY_ON_WRITE", "SPARK", "false"),
+      Arguments.of("false", "MERGE_ON_READ", "AVRO", "false"),
+      Arguments.of("false", "MERGE_ON_READ", "SPARK", "false"),
+      Arguments.of("true", "COPY_ON_WRITE", "AVRO", "false"),
+      Arguments.of("true", "COPY_ON_WRITE", "SPARK", "false"),
+      Arguments.of("true", "MERGE_ON_READ", "AVRO", "false"),
+      Arguments.of("true", "MERGE_ON_READ", "SPARK", "false"),
+      Arguments.of("false", "COPY_ON_WRITE", "AVRO", "true"),
+      Arguments.of("false", "COPY_ON_WRITE", "SPARK", "true"),
+      Arguments.of("false", "MERGE_ON_READ", "AVRO", "true"),
+      Arguments.of("false", "MERGE_ON_READ", "SPARK", "true"),
+      Arguments.of("true", "COPY_ON_WRITE", "AVRO", "true"),
+      Arguments.of("true", "COPY_ON_WRITE", "SPARK", "true"),
+      Arguments.of("true", "MERGE_ON_READ", "AVRO", "true"),
+      Arguments.of("true", "MERGE_ON_READ", "SPARK", "true"))
```

Comment:
We don't need to test position based for COW. Or for fg reader disabled. So that should cut down the combos

Reply:
good point.